iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java

```diff
@@ -90,19 +93,58 @@ private static Map<String, String> validateAndReadFiles(InputFile inputFile, Map
     Map<String, String> fileContents = new HashMap<>(files.size());
 
     for (Map.Entry<String, InputFile> filenameToInputFile : files.entrySet()) {
-      var additionalInputFile = filenameToInputFile.getValue();
-      String fileContent;
-      try {
-        fileContent = additionalInputFile.contents();
-      } catch (IOException e) {
-        throw parseExceptionFor(inputFile, "Failed to read file at " + additionalInputFile, e.getMessage());
+      if (filenameToInputFile.getKey().endsWith("tgz") || filenameToInputFile.getKey().endsWith("gz")) {
+        fileContents.putAll(readCompressedFile(filenameToInputFile.getKey(), filenameToInputFile.getValue()));
+      } else {
+        var fileContent = readTextFile(inputFile, filenameToInputFile);
+        fileContents.put(filenameToInputFile.getKey(), fileContent);
       }
-
-      fileContents.put(filenameToInputFile.getKey(), fileContent);
     }
     return fileContents;
   }
 
+  // exposed for tests
+  static Map<String, String> readCompressedFile(String name, InputFile inputFile) {
+    LOG.debug("Read dependency chart {}", name);
+    var pathPrefix = pathPrefix(name);
+    Map<String, String> result = new HashMap<>();
+    try (var source = inputFile.inputStream();
+      var gzip = new GZIPInputStream(source);
```

Comment:
What is your security strategy to prevent vulnerabilities like zip slip? What about DOS? For me, this approach is super risky and I am not happy with handling compressed files. Please check with @agigleux if this is the approach we would like to have in the analyzer.